packages/reference/src/common/queryClient.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+
+import { QueryClient, useQuery as useRQ } from '@tanstack/react-query';
+
+/**
+ * A custom client context ensures zero conflict with host apps also using
+ * React Query.
+ */
+const clientContext = React.createContext<QueryClient | undefined>(undefined);
+
+export function useQueryClient(): QueryClient {
+  const client = React.useContext(clientContext);
+
+  return React.useMemo(() => {
+    if (client) {
+      return client;
+    }
+
+    return new QueryClient({
+      defaultOptions: {
+        queries: {
+          useErrorBoundary: false,
+          refetchOnWindowFocus: false,
+          refetchOnReconnect: true,
+          refetchOnMount: false,
+          staleTime: Infinity,
+          retry: false,
+        },
+      },
+    });
+  }, [client]);
+}
+
+// @ts-expect-error
+export const useQuery: typeof useRQ = (key, fn, opt) => {
+  return useRQ(key, fn, { ...opt, context: clientContext });
```

Comment:
I don't fully understand why you need to override the `useQuery` wondering could you please explain that to me?

Reply:
It's only to pass the custom `context` without it it will crash